mapping.Rmd

```diff
@@ -31,1098 +33,1520 @@ options(scipen = 999)
-  code = c("AK", "ME", "WI", "VT", "NH", "WA", "ID", "MT", "ND", "MN", "IL", "MI", "NY", "MA", "OR", "NV", "WY", "SD", "IA", "IN", "OH", "PA", "NJ", "CT", "RI", "CA", "UT", "CO", "NE", "MO", "KY", "WV", "VA", "MD", "DE", "AZ", "NM", "KS", "AR", "TN", "NC", "SC", "DC", "OK", "LA", "MS", "AL", "GA", "HI", "TX", "FL"),
-  name = c("Alaska", "Maine", "Wisconsin", "Vermont", "New Hampshire", "Washington", "Idaho", "Montana", "North Dakota", "Minnesota", "Illinois", "Michigan", "New York", "Massachusetts", "Oregon", "Nevada", "Wyoming", "South Dakota", "Iowa", "Indiana", "Ohio", "Pennsylvania", "New Jersey", "Connecticut", "Rhode Island", "California", "Utah", "Colorado", "Nebraska", "Missouri", "Kentucky", "West Virginia", "Virginia", "Maryland", "Delaware", "Arizona", "New Mexico", "Kansas", "Arkansas", "Tennessee", " North Carolina", "South Carolina", " District of Columbia", "Oklahoma", "Louisiana", "Mississippi", "Alabama", "Georgia", "Hawaii", "Texas", "Florida")
-)
+# You can either run this once per script to automatically style all maps with
+# the Urban theme
+set_urbn_defaults(style = "map")
 
-# plot!
-usa_drivers %>%
-	ggplot(mapping = aes(x = Year, y = `Share Distracted`, group = abbreviation)) +
-  geom_line() +
-	scale_x_discrete(breaks = c(2010, 2015), labels = c("'10", "'15")) +
-	scale_y_continuous(expand = expand_scale(mult = c(0, 0.2)), breaks = c(0, 0.5, 1)) +
-  facet_geo(facets = ~abbreviation, grid = urban_grid) +
-	labs(title = "Driving is dangerous", 
-			 subtitle = "Simulated Distracted Driving is Increasing",
-			 x = NULL,
-			 y = NULL,
-			 caption = "National Highway Traffic Administration \n Data from library(fivethirtyeight) + Random Noise") +
-	theme(plot.background = element_rect(colour = "white"),
-				axis.text.x = element_text(margin = margin(t = 2)),
-  			axis.text = element_text(size = 8L),
-  			axis.line = element_blank(),
-				panel.border = element_rect(colour = "black", 
-																		fill = NA, 
-																		size = 0.3),
-        strip.background = element_rect(fill = "grey85", 
-        																colour = "black", 
-        																size = 0.3),
-				axis.ticks.length = unit(1L, "pt"),
-				strip.text.x = element_text(margin = margin(t = 1, b = 1), size = 11))
+# Or you can add `+ theme_urbn_map()` to the end of every map you make
+ggplot() +
+  geom_sf(states, mapping = aes()) +
+  theme_urbn_map()
 ```
 
-## Dot Maps
+### Layering
+
+You can layer multiple points/lines/polygons on top of each other using
+the `+` operator from `library(ggplot2)`. The shapes will appear from
+bottom to top (ie the last mapped object will show up on top). It is
+important that all layers are in the same CRS (coordinate reference
+system).
+
+```{r layers, cache = TRUE}
+
+state_capitals <- state_capitals %>%
+  # This will change CRS to ESRI:102003 and shift the AK and HI state capitals
+  # point locations to the appropriate locations on the inset maps.
+  tigris::shift_geometry()
```

Comment:
Honolulu and Juneau still look off

Reply:
Yeah and they will be until we updated urbnmapr to use tigris::shift_geometry() everywhere. Which I think shoudl be our next priority. For now I'm going to just filter out AL and HI capitals from the map